aten/src/ATen/native/Copy.cpp

```diff
@@ -114,13 +116,13 @@ void _copy_same_type__cpu(Tensor& self, const Tensor& src) {
     } else {
 #ifdef _OPENMP
       if (!in_parallel_region()) {
+        auto iter = TensorIterator::unary_op(self, src);
         AT_DISPATCH_ALL_TYPES_AND(
-          at::ScalarType::Half, self.scalar_type(), "_copy_same_type_", [&]() {
-            at::CPU_tensor_parallel_apply2<scalar_t, scalar_t>(
-                self, src, [](scalar_t& self_val, const scalar_t& src_val) {
-                  self_val = src_val;
-                });
-          });
+            at::ScalarType::Half, self.scalar_type(), "_copy_same_type_", [&] {
+              unary_kernel(
```

Comment:
Maybe supporting half and using vectorized instructions if available could improve performance? This then also requires moving this into cpu for codegen.